parser/src/main/scala/jawn/ByteArrayParser.scala

```diff
@@ -0,0 +1,24 @@
+package org.typelevel.jawn
+
+/**
+ * Basic byte array parser.
+ */
+final class ByteArrayParser[J](src: Array[Byte]) extends SyncParser[J] with ByteBasedParser[J] {
```

Comment:
Some of these are `private[jawn]` (e.g., `StringParser`), and some are public.  Is there a reason any of them are public?  Something to think about before 1.0.

Reply:
I noticed this today and thought I could remember noticing it before and finding out there was some reason for it. I'll look again before 1.0.0.